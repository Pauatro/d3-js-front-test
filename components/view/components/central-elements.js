import colors from "../../../utils/colors"
import{
    formatNumber
} from "../../../utils/formatting"

export const createCentralTitle = (svg, name, innerRadius)=>{
    svg
        .append('g')
        .append('text')
        .append('tspan')
        .attr('class', 'central-title')
        .text(name.toUpperCase())
        .attr('dy', -innerRadius/1.4)
        .attr('style', `
            font-size: ${innerRadius/5}px;
            stroke-width: 0.5px;
            fill: ${colors.lightGrey};
            stroke: ${colors.lightGrey};
            text-anchor: middle;
            dominant-baseline: middle;
            font-weight: lighter;
            letter-spacing: 1px;
        `)
}

export const createCentralData = (svg, innerRadius, total, units)=>{    
    svg
        .append('g')
        .append('text')
        .attr('width', innerRadius*2)
        .attr('height', innerRadius*2)
        .append('tspan')
        .text(formatNumber(total) + (units?units:''))
        .attr("dy", -innerRadius / 2.75)
        .attr('style', `
            font-size: ${innerRadius/3.2}px;
            text-anchor: middle;
            dominant-baseline: middle;
            stroke-width: 0.7px;
            fill: ${colors.darkGrey};
            stroke: ${colors.darkGrey};
        `)
}