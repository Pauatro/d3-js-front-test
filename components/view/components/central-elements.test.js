import createViewContainer from './view-container.js'
import {createCentralData, createCentralTitle} from './central-elements.js'


it('renders correctly', () => {

    const container = document.createElement('div')
    container.style = "width: 300px; height: 200px;"

    document.body.append(container)

    const {
        svg,
        containerHeight
    } = createViewContainer(container, 0)

    const outerRadius = containerHeight/3.5
    const innerRadius = outerRadius*0.9

    createCentralTitle(svg, 'name', innerRadius)
    expect(svg).toMatchSnapshot();
    createCentralData(svg, innerRadius, 10000, 'â‚¬')
    expect(svg).toMatchSnapshot();
});