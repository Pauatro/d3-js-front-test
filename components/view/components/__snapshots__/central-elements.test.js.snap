// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Selection {
  "_groups": Array [
    Array [
      <svg
        viewBox="0,0,0,0"
      >
        <g>
          <text>
            <tspan
              class="central-title"
              dy="0"
              style="
            font-size: 0px;
            stroke-width: 0.5px;
            fill: #b3b3b3;
            stroke: #b3b3b3;
            text-anchor: middle;
            dominant-baseline: middle;
            font-weight: lighter;
            letter-spacing: 1px;
        "
            >
              NAME
            </tspan>
          </text>
        </g>
      </svg>,
    ],
  ],
  "_parents": Array [
    null,
  ],
}
`;

exports[`renders correctly 2`] = `
Selection {
  "_groups": Array [
    Array [
      <svg
        viewBox="0,0,0,0"
      >
        <g>
          <text>
            <tspan
              class="central-title"
              dy="0"
              style="
            font-size: 0px;
            stroke-width: 0.5px;
            fill: #b3b3b3;
            stroke: #b3b3b3;
            text-anchor: middle;
            dominant-baseline: middle;
            font-weight: lighter;
            letter-spacing: 1px;
        "
            >
              NAME
            </tspan>
          </text>
        </g>
        <g>
          <text
            height="0"
            width="0"
          >
            <tspan
              dy="0"
              style="
            font-size: 0px;
            text-anchor: middle;
            dominant-baseline: middle;
            stroke-width: 0.7px;
            fill: #302c2c;
            stroke: #302c2c;
        "
            >
              10.000â‚¬
            </tspan>
          </text>
        </g>
      </svg>,
    ],
  ],
  "_parents": Array [
    null,
  ],
}
`;
